### PR TITLE
Add recipe for segyio

### PR DESCRIPTION
**Pull request by jokva**
Add recipe for https://github.com/equinor/segyio

### CODE REVIEW

File: recipes/segyio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+    - pip
+    - setuptools
+    - setuptools_scm
+    - pytest
+    - pytest-runner
+    - numpy
+  run:
+    - python
+    - numpy
```
Comment (chrisburr) [inline recipes/segyio/meta.yaml]: Is this package building against numpy's C API?

If so, this line should be `{{ pin_compatible('numpy') }}`. See [here](https://conda-forge.org/docs/meta.html#building-against-numpy) for details.

Comment (jokva) [reply recipes/segyio/meta.yaml]: No, all numpy use is from Python only.

**Comment by chrisburr** [comment]
@jokva Is this ready from your perspective? (We can ignore the `win_64` failure on Azure.)

**Comment by jokva** [comment]
Yes.

**Comment by whimian** [comment]
@chrisburr Why is this not approved?

**Comment by chrisburr** [comment]
It was approved but it looks like there was API instability at the time causing things to fail. I'll move this discussion to the issue on the feedstock: https://github.com/conda-forge/segyio-feedstock/issues/2

**Pull request merged**